numba/_devicearray.cpp

```diff
@@ -0,0 +1,117 @@
+#include "_pymodule.h"
+
+/* Include _devicearray., but make sure we don't get the definitions intended
+ * for consumers of the Device Array API.
+ */
+#define NUMBA_IN_DEVICEARRAY_CPP_
```

Comment:
I don't think I'd add `#ifndef NUMBA_IN_DEVICEARRAY_CPP_`, because that implies someone is including it somewhere, and I don't think this file should be included in anything else.

I think for similar reasons, I'd probably not undef it at the end of the file either.